src/algebra/module/composition_series.lean

```diff
@@ -0,0 +1,107 @@
+import order.jordan_holder
+import order.cover
+import ring_theory.simple_module
+
+def partial_order.bot_or_top {α : Type*} [partial_order α] (x z : α) :=
+  ∀ ⦃y⦄, x ≤ y → y ≤ z → (y = x ∨ y = z)
+
+lemma covby.bot_or_top {α : Type*} [partial_order α] {x z : α} (hxz : x ⋖ z) :
+  partial_order.bot_or_top x z := λ y hxy hyz,
+begin
+  cases lt_or_eq_of_le hxy with hxy h,
+  cases lt_or_eq_of_le hyz with hyz h,
+  { exfalso,
+    exact hxz.2 hxy hyz },
+  exact or.inr h,
+  exact or.inl h.symm
+end
```

Comment:
I think this would be better to write as
```suggestion
lemma covby.eq_or_eq_of_le_of_le {α : Type*} [partial_order α] {x y z : α} (hxz : x ⋖ z)
  (hxy : x ≤ y) (hyz : y ≤ z) : y = x ∨ y = z :=
begin
  cases lt_or_eq_of_le hxy with hxy h,
  cases lt_or_eq_of_le hyz with hyz h,
  { exfalso,
    exact hxz.2 hxy hyz },
  exact or.inr h,
  exact or.inl h.symm
end
```

Reply:
... Except this already basically exists, it's [`wcovby.eq_or_eq`](https://leanprover-community.github.io/mathlib_docs/order/cover.html#wcovby.eq_or_eq)

Reply:
What's msising is the iff version of `wcovby.eq_or_eq` and [`wcovby_of_eq_or_eq`](https://leanprover-community.github.io/mathlib_docs/order/cover.html#wcovby_of_eq_or_eq)

Reply:
#17262